sql/src/main/java/io/crate/execution/engine/collect/collectors/OrderedLuceneBatchIteratorFactory.java

```diff
@@ -102,22 +104,24 @@
             );
         }
 
-        private CompletableFuture<List<KeyIterable<ShardId, Row>>> tryFetchMore(ShardId shardId) {
+        private CompletionStage<List<KeyIterable<ShardId, Row>>> tryFetchMore(ShardId shardId) {
```

Comment:
I think here we could actually return the new `KillableCompletableFuture`. Otherwise the call-site has to know that this might implement `Killable`, that is not very common.

By changing the interface to return `CompletionStage` I was referring to `loadNextBatch`, but that could probably be a dedicated PR.

Reply:
Or maybe we expose a `KillableCompletionStage` here?

Reply:
> By changing the interface to return CompletionStage I was referring to loadNextBatch

That's already in place.